### PR TITLE
[feat] : add custom title bar

### LINKED ISSUE #1995: [FR] Windows 10 Native App Bar Does Not Respond To System Set Theme
### Description

The native menu bar on Windows 10 does not respond to user settings in the Windows system preferences.

image

image

### Impact

I could live without this.

### Additional Context

https://pub.dev/packages/bitsdojo_window

### PR DESCRIPTION
**Pull request by GouravShDev**
closes #1995 
- Added title bar in home screen using bitsdojo_window as mentioned in the issue.
- used default colors for window button apart from icon button which uses primary color
- Tested only on windows machine as I don't have other OS machines

Light Theme: 
image
Dark Theme:
image

### CODE REVIEW

**Comment by LucasXu0** [comment]
There is a class named `AppWindow` that manages the size and title of window. However, it seems that we should keep only one of the two options: either `windows_manager` or `bitsdojo_window`.

**Comment by a-wallen** [comment]
I agree @LucasXu0, but maybe in a separate PR. If this lands, I will make a separate issue.

File: frontend/appflowy_flutter/lib/workspace/presentation/home/home_screen.dart [hunk 1]
```diff
@@ -287,3 +305,62 @@ class HomeScreenStackAdaptor extends HomeStackDelegate {
...
+      appWindow.maximizeOrRestore();
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    buttonColors = WindowButtonColors(
+        normal: Colors.transparent,
+        iconNormal: Theme.of(context).colorScheme.primary,
+        mouseOver: const Color(0xFF404040),
```
Comment (a-wallen) [inline frontend/appflowy_flutter/lib/workspace/presentation/home/home_screen.dart:331]: These values should be read from our theme, not hardcoded here. If there are not in the theme, please add them as you see fit. Somewhere here: frontend/appflowy_flutter/packages/flowy_infra/lib/theme.dart

File: frontend/appflowy_flutter/linux/my_application.cc [hunk 2]
```diff
@@ -54,7 +55,9 @@ static void my_application_activate(GApplication *application)
     gtk_window_set_title(window, "AppFlowy");
   }
 
-  gtk_window_set_default_size(window, 1280, 720);
+  auto bdw = bitsdojo_window_from(window);
+  bdw->setCustomFrame(true);
+  // gtk_window_set_default_size(window, 1280, 720);
```
[the remark this answers is not in the document]
Comment (a-wallen) [reply frontend/appflowy_flutter/linux/my_application.cc:60]: If it isn't being used, then remove it please.

**Comment by a-wallen** [comment]
The integration tests failed on macOS since Flutter 3.7.5 dropped OpenGL support. Your PR was probably failing for reasons unrelated to your changes. I rebased on the commits that resolved this issue https://github.com/AppFlowy-IO/AppFlowy/issues/2200.

**Comment by Xazin** [comment]
Hey @GouravShDev If you need help let me know, I am available.

**Comment by GouravShDev** [comment]
@Xazin was busy this week will probably work on this on upcoming weekend. Thanks for asking

**Comment by GouravShDev** [comment]
- Remove windows button colors as they were just default color.
- Used colorscheme's error for close button it was close to red.
- change file_picker because it was giving compiler error when using <=5.0.0 version.

**Comment by a-wallen** [comment]
Looks good @GouravShDev, I'm wondering if it would look better if the title bar color was the same as the left navigation bar color. Do you mind sending a screenshot so that we can see what it would look like? See the image below as an example.

image

**Comment by GouravShDev** [comment]
@a-wallen sure.

Dark Mode:

image

Light Mode:

image

**Comment by GouravShDev** [comment]
Should I add applogo in title bar as it was by default previously ?

OLD:

image

NEW:

image

image

**Comment by annieappflowy** [comment]
> @a-wallen sure.
> 
> Dark Mode:
> 
> image
> 
> Light Mode:
> 
> image

@hyj1204 could you provide feedback on the design?
I personally feel one logo is enough

**Comment by hyj1204** [comment]
@annieappflowy 
I agree. I think we don't need the logo on the title bar.
@GouravShDev 
Could you help change the icon color to the **tertiary** color from the default theme? I think it would looks more consistent. Others looks good to me. 
image

**Comment by GouravShDev** [comment]
@hyj1204 sure.

**Comment by annieappflowy** [comment]
@GouravShDev , how's the progress so far? Do you need help from us? Please feel free to reach out on Discord

**Comment by GouravShDev** [comment]
> @GouravShDev , how's the progress so far? Do you need help from us? Please feel free to reach out on Discord

I have done the changes @hyj1204 asked. PR is ready to merge unless more changes are required.

**Comment by hyj1204** [comment]
I tried to run it in my macOS but it immediately quit after starting the app somehow. Could you double check if the same issue happen your macOS? @LucasXu0 

https://user-images.githubusercontent.com/14248245/234092588-3b5fdec6-4476-48a2-9dc2-a9c104f987b6.mov

**Comment by Xazin** [comment]
> I tried to run it in my macOS but it immediately quit after starting the app somehow. Could you double check if the same issue happen your macOS? @LucasXu0
> 
>  Screen.Recording.2023-04-24.at.2.06.21.PM.mov

Crashes on my MacOS as well.

**Comment by GouravShDev** [comment]
I don't have Mac so I won't be able to fix this issue. I have done native file changes for MacOS as mentioned in the documentation of the bitdojo_window plugin.
[https://pub.dev/packages/bitsdojo_window#for-macos-apps](url)

**Comment by a-wallen** [comment]
It looks like we need someone with a Mac to pick this branch up and finish the macOS side of things. 

I don't have a Mac. I will try to find someone that does on Discord.

**Comment by LucasXu0** [comment]
@hyj1204 @Xazin same here. The app crashed directly.

**Comment by a-wallen** [comment]
I'll have access to a mac next week. I can pick up this PR when that happens.

**Comment by Xazin** [comment]
> I'll have access to a mac next week. I can pick up this PR when that happens.

Let's see, I might have some time tonight or tomorrow to resolve it. It shouldn't be too hard, I think it's primarily because of `await AppWindow.initialize();` calling this right after `doWhenWindowReady`.

I'll send a link to the commit if I manage to fix it.

**Comment by Xazin** [comment]
See https://github.com/GouravShDev/AppFlowy/pull/1

CC @LucasXu0 @a-wallen @GouravShDev

**Comment by GouravShDev** [comment]
@Xazin Thanks.
I'll check for windows.

**Comment by Xazin** [comment]
> @Xazin Thanks.
> I'll check for windows.

Glad I could help.

Let me know if there are any issues.

**Comment by GouravShDev** [comment]
It's working on windows. But we might need to check for Linux as well.

**Comment by LucasXu0** [comment]
## macOS 

### main branch.
Screenshot 2023-05-11 at 15 53 01

### this PR.
Screenshot 2023-05-11 at 15 54 00


@GouravShDev, this PR should not affect the existing style for macOS. I suggest implementing a different titlebar style for macOS and Windows/Linux. The style for macOS should be frameless, while the style for Windows or Linux should be a normal titlebar style.

**Comment by a-wallen** [comment]
Since this thread became unexpectedly long I should probably mention that I appreciate everyone who has contributed to this @GouravShDev @Xazin @LucasXu0 @hyj1204  🙌

**Comment by Xazin** [comment]
@GouravShDev I'll make a PR to your branch next week to fix the MacOS title bar.

**Comment by annieappflowy** [comment]
Is this PR ready to merge? @GouravShDev

**Comment by annieappflowy** [comment]
Close this PR because no progress has been made since last month

**Pull request closed without merging**